test/dotnet-install-tool.Tests/GivenDotnetInstallTool.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using FluentAssertions;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using Xunit;
+using System.Linq;
+
+namespace Microsoft.DotNet.Cli.Install.Tests
+{
+    public class GivenDotnetInstallTool : TestBase
+    {
+        [Fact]
+        public void ItErrorsWithANonExistentSource()
+        {
+            var result = new InstallCommand()
```

Comment:
This will trigger adding PATH to CI machine.

And to use a very high level test(slow and flaky) to test error message does not worth it.

It is a indication of no good separation of error handing ("format what color what goes to print log vs error" with rest of the code like download, parsing, path handling)

Reply:
We could have one or two such high level test to **double check** basic scenarios. But high level test should be reduced much as possible.

Reply:
We actually have a good separation in production code.
You can just pass in a action => () = throw TheExceptionYouWant to installcommand, and assert on IReport

Reply:
Can you elaborate on what `trigger adding PATH to CI machine` means?

This particular test is intended to make sure the `source` option is being passed down through the various layers and not the error message itself; just the fact it errors means the source option was passed through correctly.  I think this test is poorly named and should be renamed at the very least.

This test, however, would have caught the missing line in the command passing in the source option with my original PR.

Additionally, because we're spawning another process in these tests, we can easily capture the output and examine it; this is desired for testing an option like `verbose` which changes MSBuild output.

Reply:
> Can you elaborate on what trigger adding PATH to CI machine means?

1. During first run experience, CLI will try to add .dotnet/tools to PATH. But it will not when certain environment variables are set.

2. During dotnet install, it will check and print messages.

And dependends on how to assert, due to .dotnet/tools is not present on PATH, test may behavir differently.

> This particular test is intended to make sure the source option is being passed down through the various layers and not the error message itself; just the fact it errors means the source option was passed through correctly. I think this test is poorly named and should be renamed at the very least.

> This test, however, would have caught the missing line in the command passing in the source option with my original PR.

This test gaps is fixed by https://github.com/dotnet/cli/pull/8402 . It is unfortunate it takes so long. But this should be in the same PR as InstallCommand.cs